src/metabase/driver/postgres.clj

```diff
@@ -1247,3 +1275,22 @@
                {:name "Scaleway" :pattern "\\.scw\\.cloud$"}
                {:name "Supabase" :pattern "(pooler\\.supabase\\.com|\\.supabase\\.co)$"}
                {:name "Timescale" :pattern "(\\.tsdb\\.cloud|\\.timescale\\.com)$"}]})
+
```

Comment:
I am not quite sure what is going on here. I think the intended way to insert arrays (beyond basic primitives) is to use `createArrayOf` on the connection https://jdbc.postgresql.org/documentation/server-prepare/#arrays

As this maybe messes with more than its worth - could restrict support to primitive/string arrays which _should_ be accepted by the driver as-is and do not require creating `java.sql.Array` objects.

Have I got something wrong? I might be reading this wrong?

Reply:
Ouch! Yep that makes sense!